src/ert/run_models/everest_run_model.py

```diff
@@ -307,16 +318,60 @@ def _forward_model_evaluator(
     def _get_cached_results(
         self, control_values: NDArray[np.float64], evaluator_context: EvaluatorContext
     ) -> dict[int, Any]:
-        cached_results: dict[int, Any] = {}
-        if self._simulator_cache is not None:
-            for control_idx, real_idx in enumerate(evaluator_context.realizations):
-                cached_data = self._simulator_cache.get(
-                    self._everest_config.model.realizations[real_idx],
-                    control_values[control_idx, :],
+        assert self._experiment is not None
+
+        control_groups = {c.name: c for c in self._everest_config.controls}
+        control_variables = {g: len(c.variables) for g, c in control_groups.items()}
+        control_group_spans: list[tuple[int, int]] = []
+        span_: int = 0
+        for num_vars in control_variables.values():
+            control_group_spans.append((span_, span_ + int(num_vars)))
+            span_ += int(num_vars)
+
+        def controls_1d_to_dict(values_: list[float]) -> dict[str, list[float]]:
+            return {
+                group: values_[span_[0] : span_[1]]
+                for group, span_ in zip(
+                    control_groups, control_group_spans, strict=False
                 )
-                if cached_data is not None:
-                    cached_results[control_idx] = cached_data
-        return cached_results
+            }
+
+        cached_results2: dict[int, Any] = {}
```

Comment:
Variables with numbers in them gets me a bit confused 😅 Could this be rewritten/simplified?